addons/survey/static/src/question_page/question_page_list_renderer.js

```diff
@@ -54,18 +54,27 @@ export class QuestionPageListRenderer extends ListRenderer {
         const sectionColumns = [];
 
         let titleColumnIndex = 0;
+        let number_fields = 0;
         for (const col of columns) {
             if (col.type !== "field") {
+                number_fields -= 1;
                 continue;
             }
             if (col.widget === "handle") {
+                number_fields -= 1;
                 titleColumnIndex = 1;
                 sectionColumns.push(col);
                 continue;
             }
             if (this.fieldsToShow.includes(col.name) && col.name !== this.titleField) {
                 sectionColumns.push(col);
             }
+            if(col.type === 'field'){
+                number_fields += 1;
+            }
+        }
+        if (columns.length != number_fields){ // Failsafe for testing
```

Comment:
Also to be updated.

Reply:
For this I changed the logic, instead of counting the number of fields that the view has, we check if there is originally a trash button in the row.
If this button doesn't exist in the original columns this can lead to duplicated columns and thus keys: which was the cause of the failed QUnit tests.

If you see another way to do this you can tell me.

Reply:
In the most simple case, it works. Then I thought it could be simplified if we just did this instead, also working in base conditions.
```js
if (col.type !== 'field) {
    sectionColumns.push(col);  // this guy
    continue;
}
``` 
However, in both cases, we are still missing the sequence part of the point, needed for a colspan. 
If we try and move columns (e.g., with studio), both implementations fail. It would seem to me that this colspan/section columns should be computed differently. 
Perhaps something like this, in pseudocode?
```py
title_col_idx = 0
colspan = 1
found = False
for i in range(len(columns)):
    if not found and (not col.name or col.name != this.titleField):
        continue
    if not found:
        found=True
        title_col_from_idx = i
    if col.name and col.name not in fieldsToShow:  
        # Stop to show the column.
        break
    colspan += 1

columns[title_col_idx].colspan = colspan
section_columns = columns[:, title_col_idx+1] + [title_col_idx+colspan, :]
```

Would you like to have a go at it in JS?

Reply:
I do not see why the 1st solution you proposed is not good for our problem:
```js
if (col.type !== 'field) {
    sectionColumns.push(col);  // this guy
    continue;
}
```
I see a big problem with the new colspan computation you proposed:
`columns` contains all of the fields shown for the `survey.question` model, so: 
> [sequence, title, question_type, random_questions_count, buttons]
In the current configuration of columns the new computation works.

But if the columns are in these configurations it won't work:
> [sequence, title, random_questions_count, question_type, buttons]
> [sequence, question_type, title, random_questions_count, buttons]

If, for some reason, the user wants to change the columns order in one of these, via studio or other means, question_type would be comprised in the section_columns, leading to either unwanted columns or even crashes.